opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -724,20 +724,10 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
                 if (fluidState.saturation(gasPhaseIdx) > 0.0
                     && Rs >= (1.0 - 1e-10)*oilPvt_->saturatedGasDissolutionFactor(regionIdx, Opm::scalarValue(T), Opm::scalarValue(p)))
                 {
-                    if (fluidState.saturation(gasPhaseIdx) < 1e-4) {
```

Comment:
Potential for spectacular fireworks downstream. I assume that you did test this well?